crates/sparrow-main/tests/e2e/collection_tests.rs

```diff
@@ -67,6 +71,57 @@ async fn test_get_dynamic_key() {
     "###);
 }
 
+#[tokio::test]
+async fn test_first_map() {
+    // insta::assert_snapshot!(QueryFixture::new("{ value: Input.e0 | last() }").run_to_csv(&collection_data_fixture().await).await.unwrap(), @r###"
+    // _time,_subsort,_key_hash,_key,value
+    // "###);
+
+    let hash = QueryFixture::new("{ value: Input.e0 | first() }")
+        .with_dump_dot("asdf")
+        .run_to_parquet(&collection_data_fixture().await)
+        .await
+        .unwrap();
+
+    // let expected = "adsf";
+    // assert_eq!(hash, expected);
+
+    let file = std::fs::File::open(hash).unwrap();
+    let parquet_reader = ParquetRecordBatchReaderBuilder::try_new(file)
+        .unwrap()
+        .with_batch_size(1000);
+    let reader = parquet_reader.build().unwrap();
+    let schema = reader.schema();
+    let batches: Vec<_> = reader.try_collect().unwrap();
+    let concatenated = arrow::compute::concat_batches(&schema, &batches).unwrap();
+    println!("{:?}", concatenated);
+}
```

Comment:
Verified reading the parquet file that the values are as expected! Unfortunately can't print a map in csv, but can add unit tests to do `first/last`, then a `get`.